test-common/test-common-api/src/main/java/org/apache/kafka/common/test/api/ClusterInstance.java

```diff
@@ -225,4 +235,53 @@ default void waitAcls(AclBindingFilter filter, Collection<AccessControlEntry> en
         }
     }
 
+    default void waitForTopicDeletion(String topic, int numPartitions) throws Exception {
```

Comment:
Could you please move all checks to `ClusterInstance#waitForTopic`? those checks should be enabled when the input `partitions` is equal to 0